File: articles/moral-psychology-in-organizations-and-institutions/c/unethical_probability_score.c
/*
Synthetic unethical action probability example.
Educational scaffold only.

Compile:
  cc c/unethical_probability_score.c -lm -o outputs/tables/unethical_probability_score_c
Run:
  ./outputs/tables/unethical_probability_score_c
*/

#include <math.h>
#include <stdio.h>

double logistic(double x) {
    return 1.0 / (1.0 + exp(-x));
}

int main(void) {
    double performance_pressure = 0.9;
    double hierarchy_intensity = 0.7;
    double diffusion_responsibility = 0.8;
    double accountability_strength = 0.5;
    double cultural_climate = 0.4;
    double speak_up_protection = 0.3;
    double value_alignment = 0.4;
    double corruption_pressure = 0.6;

    double latent =
        0.35 * performance_pressure +
        0.25 * hierarchy_intensity +
        0.30 * diffusion_responsibility -
        0.40 * accountability_strength -
        0.25 * cultural_climate -
        0.20 * speak_up_protection -
        0.25 * value_alignment +
        0.30 * corruption_pressure;

    double probability = logistic(latent);

    printf("Synthetic unethical action probability: %.3f\n", probability);
    return 0;
}

File: articles/moral-psychology-in-organizations-and-institutions/go/unethical_probability_score.go
// Synthetic unethical action probability example.
// Educational scaffold only.
//
// Run:
//   go run go/unethical_probability_score.go

package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func main() {
	performancePressure := 0.9
	hierarchyIntensity := 0.7
	diffusionResponsibility := 0.8
	accountabilityStrength := 0.5
	culturalClimate := 0.4
	speakUpProtection := 0.3
	valueAlignment := 0.4
	corruptionPressure := 0.6

	latent := 0.35*performancePressure +
		0.25*hierarchyIntensity +
		0.30*diffusionResponsibility -
		0.40*accountabilityStrength -
		0.25*culturalClimate -
		0.20*speakUpProtection -
		0.25*valueAlignment +
		0.30*corruptionPressure

	probability := logistic(latent)

	fmt.Printf("Synthetic unethical action probability: %.3f\n", probability)
}
